test/python/tensorflow/tf_unittest_runner.py

```diff
@@ -87,41 +93,45 @@ def main():
+                                (2 if arguments.verbose else 0))
         elapsed = time.time() - start
-        print("Testing results\nTime elapsed: ", str(
-            timedelta(seconds=elapsed)))
+        print("\n\nTesting results\nTime elapsed: ",
+              str(timedelta(seconds=elapsed)))
         return check_and_print_summary(test_results, test_list[1])
 
     if (arguments.run_tests_from_file):
         all_test_list = []
         invalid_list = []
         start = time.time()
-        list_of_tests = read_tests_from_file(arguments.run_tests_from_file)
-        for test in list_of_tests:
-            test_list = get_test_list(arguments.tensorflow_path, test)
-            for test in test_list[1]:
-                if test is not None:
-                    invalid_list.append(test_list[1])
-                    result_str = "\033[91m INVALID \033[0m " + test + \
-                    '\033[91m' + '\033[0m'
-                    print('TEST:', result_str)
-            test_list = list(set(test_list[0]))
-            for test_name in test_list:
-                if test_name not in all_test_list:
-                    all_test_list.append(test_name)
-        test_results = run_test(all_test_list, xml_report)
+        list_of_tests = read_tests_from_manifest(arguments.run_tests_from_file,
+                                                 arguments.tensorflow_path)
+        # list_of_tests = list(dict.fromkeys(list_of_tests))  # remove dups
+        # for test in list_of_tests:
+        #     test_list = get_test_list(arguments.tensorflow_path, test)
+        #     for test in test_list[1]:
+        #         if test is not None:
+        #             invalid_list.append(test_list[1])
+        #             result_str = "\033[91m INVALID \033[0m " + test + \
+        #             '\033[91m' + '\033[0m'
+        #             print('TEST:', result_str)
+        #     test_list = list(set(test_list[0]))
+        #     for test_name in test_list:
+        #         if test_name not in all_test_list:
+        #             all_test_list.append(test_name)
+        test_results = run_test(list_of_tests, xml_report,
+                                (2 if arguments.verbose else 0))
         elapsed = time.time() - start
-        print("Testing results\nTime elapsed: ", str(
-            timedelta(seconds=elapsed)))
+        print("\n\nTesting results\nTime elapsed: ",
+              str(timedelta(seconds=elapsed)))
         return check_and_print_summary(test_results, invalid_list)
 
 
 def get_test_list(tf_path, test_regex):
     accepted_formats = [
-        "math_ops_test.DivNoNanTest.testBasic", "math_ops_test.DivNoNanTest.*",
-        "math_ops_test.D*", "math_ops_test.*", "math_*_test", "math_*_*_test",
-        "math*_test"
+        "*test*", "math_ops_test.DivNoNanTest.testBasic",
+        "math_ops_test.DivNoNanTest.*", "math_ops_test.D*", "math_ops_test.*",
+        "math_*_test", "math_*_*_test", "math*_test"
```

Comment:
This is just a standard regex; no need to be explicit about it.

Reply:
Dots are treated literally, not like python regex